src/assets/Standards/CollectibleStandards/ERC1155/ERC1155Standard.ts

```diff
@@ -1,45 +1,69 @@
-const ERC1155_METADATA_URI_INTERFACE_ID = '0x0e89341c';
-const ERC1155_TOKEN_RECEIVER_INTERFACE_ID = '0x4e2312e0';
+import { abiERC1155 } from '@metamask/metamask-eth-abis';
+import {
+  ERC1155,
+  ERC1155_INTERFACE_ID,
+  ERC1155_METADATA_URI_INTERFACE_ID,
+  ERC1155_TOKEN_RECEIVER_INTERFACE_ID,
+} from '../../../../constants';
 
 export class ERC1155Standard {
+  private web3: any;
+
+  constructor(web3?: any) {
+    this.web3 = web3;
```

Comment:
Can we use a type for this instead of `any`? And why is `web3` optional? It seems that everything in this class relies on `web3` being provided.

Reply:
I agree with you @mcmire, but I think for this we have to upgrade the web3 dependence, which could be done in another PR. But I'm not against this if @adonesky1 wants to upgrade the module in this PR

Reply:
Oh, does `web3` provide types now? I agree that we probably wouldn't want to upgrade it in this PR, but I was thinking in the meantime we could give this property a type on the spot based on how we end up using it in this class. For instance:

``` typescript
type ERC1155ABI = ...;
type Address = `0x${string}`

type Contract = {
  at: Address
}

type Web3 = {
  eth: {
    contract: (abi: ERC1155ABI): Contract
  }
}
```

That said, I don't believe that `abiERC1155` in `@metamask/metamask-eth-abis` has the right type on it. My feeling is that all of the objects that that module exports should have `as const` on them. Right now it's possible to pass an ABI that has the same shape but isn't _exactly_ the ERC-1155 ABI. Maybe we could have a generic ABI type to get around this, but then we'd be repeating work that is probably elsewhere. So maybe this is fine for now.

Reply:
Can this be logged somewhere for tech debt purposes, though?

Reply:
Thats a good point, I think that could work in the meantime for this PR. For future development we should upgrade this module and use the types provided by `web3` or  start using the `ethers` dependence. I remember that extension changed from `web3` to `ethers`, would be easier if we all use the same.

Reply:
Yeah, I am definitely a fan of using `ethers`. It feels a lot more maintained at this point and is fully typed, and I think we can avoid the pain that we've been experiencing with `web3`. Of course that's a larger change but that'd be my recommendation for new code going forward.

Reply:
I've made updates here: 8effcaf and
here c9abc81

Reply:
@mcmire @gantunesr take a look and let me know what you think

Reply:
Looks good, thank you!